Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs

```diff
@@ -17,11 +17,43 @@ namespace Microsoft.Build.Unity.ProjectGeneration
 {
     public class MSBuildToolsConfig
     {
+        private const int CurrentConfigVersion = 3;
+
         private static string MSBuildSettingsFilePath { get; } = Path.Combine(Utilities.ProjectPath, "MSBuild", "settings.json");
 
+        [SerializeField]
+        private int version = 0;
+
         [SerializeField]
         private bool autoGenerateEnabled = false;
 
+        [SerializeField]
+        private string dependenciesProjectGuid = Guid.NewGuid().ToString();
```

Comment:
>= Guid.NewGuid().ToString(); [](start = 47, length = 28)

Similar to the previous comment - if this class isn't actually being created/managed by Unity it doesn't seem like you need to initialize all these fields, since you already ensure their initialization in your Load code path.

Reply:
Responded to comment above, and here, the way Unity deserialization works for FromJsonOverwrite is that it won't overwrite all fields. So if any of the fields are new, and you bumped the version, you can then proceed to save with the default initialized values.

---
In reply to: [364906480](https://github.com/microsoft/MSBuildForUnity/pull/96#discussion_r364906480) [](ancestors = 364906480)